app/blueprints/base/views.py

```diff
@@ -139,6 +136,9 @@ def oidc_callback():
 
     if 'next_url' in session:
         next_url = session['next_url']
+        if next_url == 'admin/users':
+            session["reauthenticated"] = True
```

Comment:
Don't hardcode paths.

The fact that the `next_url` session variable is `admin/users` doesn't really indicate that a reauthentication has happened. A better approach (more broadly applicable than this example) would be:
 * generate a random token before requesting reauthentication
 * store the token in the session
 * send the token with the authentication request in the `state` parameter - this is passed back unchanged to the `oidc_callback`
 * compare the `state` value with the token in the session - if they match, then the reauthentication has happened